src/v/cluster/partition.cc

```diff
@@ -1236,11 +1269,10 @@ partition::sync_kafka_start_offset_override(
         }
         auto start_kafka_offset
           = _archival_meta_stm->manifest().get_start_kafka_offset_override();
-        if (start_kafka_offset != kafka::offset{}) {
-            co_return kafka::offset_cast(start_kafka_offset);
-        }
+        _cached_kafka_start_override = kafka::offset_cast(start_kafka_offset);
```

Comment:
Doesn't this mean that we'll never do the archival sync after we've cached a start override?

Reply:
Yes, that's the hope.

The log eviction stm is now caching the kafka offset in every prefix truncate batch that is being applied against it; even if they don't land in local storage. So the only cases where we'd need to get the offset override from the archival stm are where there hasn't been any prefix truncate batches applied to the log eviction stm since the RP broker started. In these cases we only need to sync the archival stm once to get the most up-to-date prefix truncation from cloud storage. And then after that if any additional prefix truncate batches come in the log eviction stm will be aware of them.